modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java

```diff
@@ -477,77 +543,70 @@ public VendorHostedDeployedServiceDetails getVendorHostedServiceDetailsByIdForEn
-        DeployTask task = new DeployTask();
-        task.setId(UUID.fromString(taskId));
-        task.setOcl(deployServiceEntity.getDeployRequest().getOcl());
-        task.setDeployRequest(deployServiceEntity.getDeployRequest());
-        fillHandler(task);
-        return task;
+    private DeployResult handlerCallbackDeployResult(TerraformResult result) {
+        DeployResult deployResult = new DeployResult();
+        if (result.getCommandSuccessful()) {
+            deployResult.setState(TerraformExecState.DEPLOY_SUCCESS);
+        } else {
+            deployResult.setState(TerraformExecState.DEPLOY_FAILED);
+            deployResult.setMessage(result.getCommandStdError());
+        }
+        deployResult.getPrivateProperties().put(STATE_FILE_NAME, result.getTerraformState());
+        if (Objects.nonNull(result.getImportantFileContentMap())) {
+            deployResult.getPrivateProperties().putAll(result.getImportantFileContentMap());
+        }
+        return deployResult;
     }
 
     /**
      * Callback method after the service is destroyed.
      */
     public void destroyCallback(String taskId, TerraformResult result) {
-        DeployServiceEntity deployServiceEntity =
-                deployServiceStorage.findDeployServiceById(UUID.fromString(taskId));
-        if (Objects.isNull(deployServiceEntity)) {
-            String errorMsg = String.format("Service with id %s not found.", taskId);
-            log.error(errorMsg);
-            throw new ServiceNotDeployedException(errorMsg);
+        DeployServiceEntity deployServiceEntity = getDeployServiceEntity(UUID.fromString(taskId));
+        DeployResult destroyResult = handlerCallbackDestroyResult(result);
+        destroyResult.setId(UUID.fromString(taskId));
+        if (StringUtils.isNotBlank(result.getTerraformState())) {
+            getResourceHandler(deployServiceEntity.getCsp()).handler(destroyResult);
         }
-        if (Boolean.TRUE.equals(result.getCommandSuccessful())) {
-            deployServiceEntity.setServiceDeploymentState(ServiceDeploymentState.DESTROY_SUCCESS);
-            deployServiceEntity.setProperties(new HashMap<>());
-            deployServiceEntity.setPrivateProperties(new HashMap<>());
-            deployResourceStorage.deleteByDeployServiceId(deployServiceEntity.getId());
-        } else {
-            deployServiceEntity.setServiceDeploymentState(ServiceDeploymentState.DESTROY_FAILED);
+        DeployServiceEntity updatedDeployServiceEntity =
+                flushDestroyServiceEntity(destroyResult, deployServiceEntity);
+        if (ServiceDeploymentState.DESTROY_SUCCESS
+                == updatedDeployServiceEntity.getServiceDeploymentState()) {
+            Deployment deployment = getDeployment(updatedDeployServiceEntity.getDeployRequest()
+                    .getOcl().getDeployment().getKind());
+            deployment.deleteTaskWorkspace(taskId);
```

Comment:
in case of terraform-boot, the workspace deletion is handled by terraform-boot. Do we really need this here?

Reply:
It's  unnecessary and I removed it.